### ISSUE TITLE
JSONEncoder and JSONDecoder conflicts

### ISSUE THREAD

**Issue by CocoaPriest** [OpenKitten/Cheetah#3]
`JSONEncoder` and `JSONDecoder` classes should be renamed because they conflict with Foundation in Swift 4:

```
error: cannot invoke 'JSONDecoder' with no arguments:
let decoder = JSONDecoder()
```

**Comment by Joannis** [comment]
It's not possible to rename these without a major increase. I approach these by specifying the Module `Foundation.JSONEncoder` and `Cheetah.JSONEncoder`

**Comment by CocoaPriest** [comment]
yes, I do the same. But imo this is not a good solution for the long term.
